builtin/readline_osh.py

```diff
@@ -142,9 +184,10 @@ def Run(self, cmd_val):
                     readline.unbind_keyseq(arg.r)
 
                 if arg.x is not None:
-                    self.errfmt.Print_("warning: bind -x isn't implemented",
-                                       blame_loc=cmd_val.arg_locs[0])
-                    return 1
+                    if self.bindx_cb is None:
```

Comment:
How about initializing `bindx_cb` in `core/shell.py`, then passing it to the constructor, and then make it a member?

That is generally the style we use

Reply:
OK, I took a closer look at this. I wrote it so that the callback's initialized only on demand; since I'd bet the vast majority of users will never use the functionality at all, why pay the price?

But the rest of oils doesn't seem to do that afaict, so I'll do as you suggested. LMK if you want me to revert it tho.

Reply:
Done.